### DISCUSSION TITLE
vscode: changes to settings are not persisted

### DISCUSSION THREAD

**Discussion by GreatBigWhiteWorld** [netblue30/firejail#5019] [Q&A] [answered]
After using vscode for a few weeks I realized that I had not used firejail to launch vscode.

Today I began vscode with "firejail code", and noticed that all the settings I had for vscode was lost. It started fresh.
I said OK no big deal I'll start fresh then.

What's strange is that later I learned that each time I launch vscode with firejail it welcomes me like the first time I use the program. All settings, extensions installed in the previous sessions (with firejail launch) are lost.

**Comment by ghost** [comment]
> All settings, extensions installed in the previous sessions (with firejail launch) are lost.

That shouldn't happen. Looking at the vscodium and code profiles I can't see anything that might be causing this. But I'm not at all familiar with the app in question. Did you use firecfg at all? If so, the correct command to start it is simply `code` (as the symlink in /usr/local/bin/code should point to firejail). Can you post the path where vscode expects its configuration on your system?

Please also post your OS, firejail version and output from:

```console
$ firejail --ignore=quiet /usr/bin/code

```

**Reply by GreatBigWhiteWorld** [reply]
```
firejail --ignore=quiet /usr/bin/code
Reading profile /etc/firejail/code.profile
Reading profile /etc/firejail/code.local
Reading profile /etc/firejail/allow-common-devel.inc
Reading profile /etc/firejail/electron.profile
Reading profile /etc/firejail/disable-common.inc
Reading profile /etc/firejail/disable-programs.inc
Parent pid 31076, child pid 31077
Warning: NVIDIA card detected, nogroups command ignored
Warning: NVIDIA card detected, nogroups command ignored
Warning: NVIDIA card detected, nogroups command ignored
Warning: NVIDIA card detected, nogroups command ignored
Child process initialized in 49.55 ms

Parent is shutting down, bye...
```
The above is when I launch code, change some settings and install an extension. Then after I close it and start it again, everyting starts new with vscode welcome dialogue like it's the first time I run it.

opensuse 15.3 
firejail version 0.9.68
code.local is an empty file when testing.
vscode config file is under ${HOME}/.config/Code

**Reply by ghost** [reply, marked as answer]
> code.local is an empty file when testing.

If you don't need it, you can safely remove it to avoid forgetting about having one.

> vscode config file is under ${HOME}/.config/Code

That path looks okay and is handled in /etc/firejail/code.profile (at least it should be).

IMO you're hit by the same bug as reported in #5012.

**Reply by GreatBigWhiteWorld** [reply]
> 
Seems like so. I did have to white list a directory in my home folder for vscode to access. I was not sure if it was expected to exclude home folder by default.
